src/components/RandomQuote.js

```diff
@@ -0,0 +1,47 @@
+import React, {Component} from 'react';
+import {connect} from 'react-redux';
+import {newQuote} from '../actions';
+import quotes from '../data/quotes.json';
+import '../styles/RandomQuote.css';
+import { FaQuoteLeft, FaTwitter } from "react-icons/fa";
+
+  class RandomQuote extends Component {
+  // use componentDidMount to get data
+  componentDidMount(){
+      this.props.dispatch(newQuote(this.getQuote()))
+  }
+  //get random quote
+  getQuote(){
+    return quotes[Math.floor(Math.random()*quotes.length)]
+  } 
+  //get new quote funtion when click button
+  getNewQuote=()=>{
+    this.props.dispatch(newQuote(this.getQuote()))
+  }
```

Comment:
You can define a function called mapDispatchToProps() that receives the dispatch() method and returns callback props that you want to inject into the component.